scripts/content_schema_artifacts.py

```diff
@@ -0,0 +1,143 @@
+import json
+import os
+from pathlib import Path
+
+from referencing import Registry, Resource, Specification
+from referencing._core import Resolver
+from referencing.jsonschema import DRAFT202012
+from referencing.typing import URI
+from script_utils.cli import run
+
+HERE = Path(__file__).parent.resolve()
+SCHEMAS = HERE.parent / "src" / "content_schemas"
+SCHEMA_ARTIFACTS = HERE.parent / "src" / "content_schema_artifacts"
+
+
+def retrieve_from_filesystem(path: Path) -> Resource:
+    """
+    Retrieve a JSON object from a file.
+
+    Args:
+        path (Path): The path to the JSON file.
+
+    Returns:
+        Resource: A resource object representing the JSON content.
+
+    Raises:
+        FileNotFoundError: If the specified file does not exist.
+    """
+    if not path.exists():
+        raise FileNotFoundError(f"Schema file '{path}' not found")
+    with path.open("r", encoding="utf-8") as file:
+        contents = json.load(file)
+    return Resource(contents=contents, specification=DRAFT202012)
```

Comment:
Why are you overriding the specification here and not using `Resource.from_contents`?

Reply:
`Resource.from_contents` cannot determine the specification since it cannot interpret that information from our content schemas. Calling it as a class method raises `CannotDetermineSpecification` (https://github.com/python-jsonschema/referencing/blob/c0b176fd62e04018aff2872a42e720d8f2caf470/referencing/_core.py#L118).

Reply:
OK, I didn't realise you generated the schemas without `$schema`. The original large schema coming out of `gen-json-schema` has it (it generates `draft-07` btw, so it might not comply with 202012).

Reply:
Ah, switched it to `draft-07`. Thanks!